File: examples/matlab/add_node_metadata_categorical.m
% define a couple edges
edges = [...
    1,2;
    2,3;...
    ];
% place the edges in a webweb struct called ww
ww.networks.network.edgeList = edges;

% Define categorical metadata sets in TWO different ways
% 1. Categories defined by integers with a set of integer-assigned keys
alphabeticallity = [1,2,3];
alphabeticallity_keys = {'A-Z','a-z','W+;'};
ww.display.metadata.alphabeticallity.values = alphabeticallity;
ww.display.metadata.alphabeticallity.categories = alphabeticallity_keys;
% 2. Categories defined directly by strings
cooperativity = {'high','medium','low'};
ww.display.metadata.cooperativity.values = cooperativity;

% BONUS: ask webweb to default to present colors by alphabeticallity
% This assignment simply needs to match the key of the metadata above.
ww.display.colorBy = 'alphabeticallity';

% call webweb
webweb(ww);

File: examples/python/add_node_metadata_categorical.py
from webweb import Web

web = Web(
    adjacency=[[0, 1], [1, 2]],
    display={
        'nodes' : {
            0 : {
                'cooperativity' : 'high',
                'alphabeticallity' : 0,
            },
            1 : {
                'cooperativity' : 'low',
                'alphabeticallity' : 1,
            },
            2 : {
                'cooperativity' : 'medium',
                'alphabeticallity' : 2,
            },
        },
        'metadata' : {
            'alphabeticallity' : {
                'categories' : ['A-Z', 'a-z', 'W+'],
            }
        },
    },
)

# we'll compute node color by the `alphabeticallity` metadata attribute
# (categorical metadata can't be used to compute node sizes)
web.display.colorBy = 'alphabeticallity'

# show the visualization
web.show()
